File: app/src/main/java/com.example.safehome/Sensors/WindowSensorFactory.java
package com.example.safehome.Sensors;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class WindowSensorFactory implements SensorFactory<Sensor> {
    private static final Logger log = LoggerFactory.getLogger(WindowSensorFactory.class);
    /**
     * FactoryMethod returns class of type Sensor.
     *
     * @param request
     * @return FactoryMethod returns class of type Sensor.
     */
    @Override
    public Sensor factoryMethod(Sensor request) {
        return null;
    }
}

File: app/src/main/kotlin/com.example.safehome/Sensors/WindowSensorFactory.kt
package com.example.safehome.Sensors

import android.hardware.Sensor
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class WindowSensorFactory : SensorFactory<Sensor?> {
    /**
     * FactoryMethod returns class of type Sensor.
     *
     * @param request
     * @return FactoryMethod returns class of type Sensor.
     */
    override fun factoryMethod(request: Sensor?): Sensor? {
        return null
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(WindowSensorFactory::class.java)
    }
}